meta(changelog): Update changelog for 8.35.0 (#14022)
getsentry/sentry-javascript: CHANGELOG.md

- "You miss 100 percent of the chances you don't take. — Wayne Gretzky" — Michael Scott

## 8.35.0

### Beta release of the official Nuxt Sentry SDK

This release marks the beta release of the `@sentry/nuxt` Sentry SDK. For details on how to use it, check out the
[Sentry Nuxt SDK README](https://github.com/getsentry/sentry-javascript/tree/develop/packages/nuxt). Please reach out on
[GitHub](https://github.com/getsentry/sentry-javascript/issues/new/choose) if you have any feedback or concerns.

- **feat(nuxt): Make dynamic import() wrapping default
  ([#13958](https://github.com/getsentry/sentry-javascript/pull/13958))** (BREAKING)
- **feat(nuxt): Add Rollup plugin to wrap server entry with `import()`
  ([#13945](https://github.com/getsentry/sentry-javascript/pull/13945))**

> There is no need to add the Node `--import` flag anymore because the SDK now wraps your application with a dynamic
> `import()`. Please update your start command to avoid initializing Sentry twice (BREAKING CHANGE). If you prefer to use
> the `--import` flag, you can enable this option in your `nuxt.config.ts`:

Reviewer: ```suggestion
**It is no longer required to add a Node `--import` flag. Please update your start command to avoid initializing Sentry twice (BREAKING CHANGE).** The SDK will now apply modifications during the build of your application to allow for patching of libraries during runtime. If run into issues with this change, you can disable this behavior in your `nuxt.config.ts` and use the `--import` flag instead:
```

```js
sentry: {
  dynamicImportForServerEntry: false;
}
```

- **feat(nuxt): Respect user-provided source map generation settings
  ([#14020](https://github.com/getsentry/sentry-javascript/pull/14020))**

As Nuxt sets `sourcemap.client: false`, it's not possible to determine whether this setting was set by you or the
framework. Set this manually like this:

> 
> ```js
> export default defineNuxtConfig({
>   sourcemap: {
>     client: true,
>   },
> });
> ```

Reviewer: ```suggestion
We now require you to explicitly enable sourcemaps for the clientside so that Sentry can unminify your erros. We made this change so source maps aren't accidentally leaked to the public. You can do so as follows:

```js
export default defineNuxtConfig({
  sourcemap: {
    client: true,
  },
});
```
```

Author: What do you think of this: 

We now respect your source map settings if they are defined, so source maps aren't accidentally leaked to the public. Since Nuxt defaults `sourcemap.client` to `false`, we can't tell if this was set by you or the framework. We now require you to explicitly enable sourcemaps for client-side source maps for Sentry can un-minify your errors:

Reviewer: Assume you don't know what sourcemaps are, or what nuxt is doing wrt sourcemaps. This is the audience we should write the changelog entry for.

Author: From this standpoint, your suggestion makes sense. I'll change that 👍

```js
export default defineNuxtConfig({
  sourcemap: {
    client: true,
  },
});
```

- feat(nuxt): Log server instrumentation might not work in dev
  ([#14021](https://github.com/getsentry/sentry-javascript/pull/14021))
- feat(nuxt): Add Http `responseHook` with `waitUntil`
  ([#13986](https://github.com/getsentry/sentry-javascript/pull/13986))

### Important Changes

- **feat(vue): Add Pinia plugin ([#13841](https://github.com/getsentry/sentry-javascript/pull/13841))**

Support for [Pinia](https://pinia.vuejs.org/) is added in this release for `@sentry/vue`. To capture Pinia state data,
add `createSentryPiniaPlugin()` to your Pinia store:

```javascript
import { createPinia } from 'pinia';
import { createSentryPiniaPlugin } from '@sentry/vue';

const pinia = createPinia();

pinia.use(createSentryPiniaPlugin());
```

- **feat(node): Implement Sentry-specific http instrumentation
  ([#13763](https://github.com/getsentry/sentry-javascript/pull/13763))**

This change introduces a new `SentryHttpInstrumentation` to handle non-span related HTTP instrumentation, allowing it to

> run side-by-side with OTel's `httpInstrumentation`. This improves support for custom OTel setups and avoids conflicts

Reviewer: ```suggestion
run side-by-side with OTel's `HttpInstrumentation`. This improves support for custom OTel setups and avoids conflicts
```

```

- **feat(node): Implement Sentry-specific http instrumentation
  ([#13763](https://github.com/getsentry/sentry-javascript/pull/13763))**

This change introduces a new `SentryHttpInstrumentation` to handle non-span related HTTP instrumentation, allowing it to
run side-by-side with OTel's `httpInstrumentation`. This improves support for custom OTel setups and avoids conflicts
with Sentry's instrumentation. Additionally, the `spans: false` option is reintroduced for `httpIntegration` to disable
span emission while still allowing custom `HttpInstrumentation` instances (`httpIntegration({ spans: false })`).

- **feat(core): Make stream instrumentation opt-in
  ([#13951](https://github.com/getsentry/sentry-javascript/pull/13951))**

This change adds a new option `trackFetchStreamPerformance` to the browser tracing integration. Only when set to `true`,
Sentry will instrument streams via fetch.

### Other Changes

- feat(node): Expose `suppressTracing` API ([#13875](https://github.com/getsentry/sentry-javascript/pull/13875))
- feat(replay): Do not log "timeout while trying to read resp body" as exception
  ([#13965](https://github.com/getsentry/sentry-javascript/pull/13965))

> - chore(deps): bump astro from 4.1.1 to 4.16.1 in /dev-packages/e2e-tests/test-applications/cloudflare-astro

Reviewer 2: Should we remove these?

Author: Yeah true, the version bumps for test-related stuff is not relevant here.
